rakelib/github_stats.rake

```diff
@@ -25,21 +28,22 @@ namespace :github_stats do
         # parse vals from json
         user = pr['user']['login']
         number = pr['number']
-        pr_created_at = DateTime.parse(pr['created_at']).to_f
+        pr_created_at = DateTime.parse(pr['created_at'])
         pr_url = pr['url']
         reviews_url = "#{pr_url}/reviews"
         # determine first_reviewed date
         uri = URI(reviews_url)
         resp = Net::HTTP.get(uri)
         first = JSON.parse(resp).first
-        first_reviewed = if first.nil?
-                           # PR has not yet been reviewed so use current timestamp
-                           DateTime.now.to_f
-                         else
-                           DateTime.parse(first['submitted_at']).to_f
-                         end
-        # calculate the duration
-        duration = (first_reviewed - pr_created_at).round
+        first_reviewed_at = if first.nil?
+                              # PR has not yet been reviewed so use current timestamp
+                              # and set time zone offset to zero so we don't get negatives
+                              DateTime.now.new_offset('+0')
```

Comment:
I did look at at the business_hours gem but it didn't seem to do quite what I needed it to do. As for the offset, the timestamps that are returned by the Github API have zero offset but when doing DateTime.now it will include the timezone offset so there is the chance of ending up with negative numbers. Setting the offset to zero gives you the proper timestamp format for calculating the correct duration.